apis/src/components/organisms/analysis/tree.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::VecDeque;
+
+use crate::providers::game_state::{GameState, GameStateSignal};
+use hive_lib::{GameType, History, State};
+use leptos::*;
+use serde::{Deserialize, Serialize};
+use tree_ds::prelude::{Node, Tree};
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct TreeNode {
+    pub turn: usize,
+    pub piece: String,
+    pub position: String,
+}
+
+#[derive(Clone)]
+pub struct AnalysisSignal(pub RwSignal<Option<AnalysisTree>>);
+#[derive(Clone, Default, Serialize, Deserialize)]
+pub struct AnalysisTree {
+    pub current_node: Option<usize>, // The node that is currently being viewed
+    pub tree: Tree<usize, TreeNode>,
+}
+
+impl AnalysisTree {
+    pub fn from_state(game_state: GameStateSignal) -> Self {
+        let gs = game_state.signal.get_untracked();
+        let mut tree = Tree::new(Some("analysis"));
+        for (i, m) in gs.state.history.moves.iter().enumerate() {
+            let parent_id = if i == 0 { None } else { Some(i - 1) };
+            let node = TreeNode {
+                turn: i,
+                piece: m.0.clone(),
+                position: m.1.clone(),
+            };
+            tree.add_node(Node::new(i, Some(node)), parent_id.as_ref())
+                .unwrap();
+        }
+        let node = if !gs.state.history.moves.is_empty() {
+            Some(gs.state.history.moves.len() - 1)
+        } else {
+            None
+        };
+        Self {
+            current_node: node,
+            tree,
+        }
+    }
+    pub fn update_node(&mut self, node: usize) {
+        let mut moves: VecDeque<(String, String)> = VecDeque::new();
+        let mut tmp_node = node;
+        loop {
+            let node = self.tree.get_node(&tmp_node).unwrap().get_value().unwrap();
+            moves.push_front((node.piece.clone(), node.position.clone()));
+            if let Some(parent) = self.tree.get_node(&tmp_node).unwrap().get_parent() {
+                tmp_node = parent
+            } else {
+                break;
+            }
+        }
+        let mut new_history = History::new();
+        new_history.game_type = GameType::MLP;
```

Comment:
Just curious but what happens when you go to analysis from a non `MLP` game?

Reply:
Figured analysis was useful wih the complete set anyways, if a player may not wish to use it, then they dont

Reply:
I think what felix means is, that if you play a non-plm game and then want to analyze it, clicking the analysis button will take you to an analysis board that has the expansion pieces. 
Which you will complain about in #serious-bug-reports

Reply:
I suggest to educate the users on this functionality this way analysis is more complete otherwise it will be a bit more cumbersome to analyze a non plm game with plm, and this is not a bug per se